pandas/tests/groupby/transform/test_transform.py

```diff
@@ -1025,7 +1025,9 @@ def test_groupby_transform_with_datetimes(func, values):
     dates = pd.date_range("1/1/2011", periods=10, freq="D")
 
     stocks = pd.DataFrame({"price": np.arange(10.0)}, index=dates)
-    stocks["week_id"] = pd.to_datetime(stocks.index).week
+    stocks["week_id"] = (
+        pd.to_datetime(stocks.index).isocalendar().set_index(stocks.index).week
+    )
```

Comment:
This is the cumbersome syntax I referenced in the PR message.

Reply:
`pd.to_datetime(stocks.index)` is just `dates`, ditto `stocks.index`, so this can become `dates.isocalendar().set_index(dates)["week"]`